test/src/code_field/code_controller_tab_test.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+import 'package:flutter_code_editor/src/symbols.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/languages/go.dart';
+
+void main() {
+  const textWithTabs = '''
+public class MyClass {
+\tpublic void main() {\t\t
+\t}\t
+}\t
+\t
+''';
+
+  group('Tab replacement', () {
+    test(
+      'applied if TabModifier is present',
+      () {
+        final controller = CodeController(
+          text: textWithTabs,
+          language: go,
+          modifiers: [const TabModifier()],
+        );
+        expect(controller.fullText.contains(Symbols.tab), false);
```

Comment:
It's better to test for the full match with the string with spaces.
Also test with different number of spaces per tab.